src/main/java/uk/ac/ebi/ampt2d/accession/study/StudyAccessioningService.java

```diff
@@ -0,0 +1,37 @@
+/*
+ *
+ * Copyright 2018 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package uk.ac.ebi.ampt2d.accession.study;
+
+import uk.ac.ebi.ampt2d.accession.common.generators.SingleAccessionGenerator;
+import uk.ac.ebi.ampt2d.accession.common.accessioning.BasicAccessioningService;
+import uk.ac.ebi.ampt2d.accession.common.persistence.DatabaseService;
+import uk.ac.ebi.ampt2d.accession.common.hashing.SHA1HashingFunction;
+
+public class StudyAccessioningService extends BasicAccessioningService<StudyModel, String, String> {
+
+    public StudyAccessioningService(DatabaseService<StudyModel, String, String> dbService) {
+        super(
+                SingleAccessionGenerator.ofSHA1AccessionGenerator(new StudyModelDigestFunction()),
+                dbService,
+                new StudyModelDigestFunction(),
```

Comment:
I find a bit confusing passing here twice the `StudyModelDigestFunction`, once to the BasicAccessioningService and once to the AccessionGenerator factory, and both do the pattern of `digestFunction.then(hashingFunction)`. maybe that's correct because you didn't want to overengineer it, but I find it a bit surprising to have the same behaviour in several places.

Reply:
They are for two different things, one is for the hashed message, the other for the accession, In this case both use the same functions but that is not necessarily true always.

Reply:
Aditionally in the case of the accession that only happens if it is gonna be an accession that is generated hashing the original object.